lib/nats.js

```diff
@@ -1166,9 +1166,16 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
         opt_callback = subject;
         subject = undefined;
     }
-    if (!msg) {
-        msg = EMPTY;
+
```

Comment:
this would be simpler:

```
    if(!this.options.json) {
        msg = !msg ? EMPTY : msg;
    } else {
        // undefined is not a valid JSON-serializable value, but null is
        msg = msg === undefined ? null : msg;
    }
```